src/filters/array.rs

```diff
@@ -0,0 +1,117 @@
+/// Filters operating on string
+use std::collections::HashMap;
+
+use serde_json::value::{Value, to_value};
+use context::JsonRender;
+use errors::TeraResult;
+use errors::TeraError;
+
+pub fn first(value: Value, _: HashMap<String, Value>) -> TeraResult<Value> {
+    let arr = try_get_value!("first", "value", Vec<Value>, value);
+
+    if let Some(val) = arr.first() {
+        Ok(val.clone())
+    } else {
+        Err(TeraError::EmptyArray("first".to_string()))
+    }
+}
+
+pub fn last(value: Value, _: HashMap<String, Value>) -> TeraResult<Value> {
+    let arr = try_get_value!("last","value",Vec<Value>, value);
+
+    if let Some(val) = arr.last() {
+        Ok(val.clone())
+    } else {
+        Err(TeraError::EmptyArray("last".to_string()))
+    }
+}
+
+pub fn join(value: Value, args: HashMap<String, Value>) -> TeraResult<Value> {
+    let arr = try_get_value!("join","value",Vec<Value>, value);
+    let separator = match args.get("d") {
+        Some(val) => try_get_value!("truncate", "d", String, val.clone()),
+        None => "".to_string(),
+    };
+    let mut reduction = arr.iter()
+        .fold("".to_string(),
+              |acc, val| acc + &*separator + &*val.render());
+    if reduction.len() > 0 {
+        // if we had any elements, trim the separator off the front.
+        reduction = reduction.split_at(separator.len()).1.to_string();
+    }
```

Comment:
`Vec` have a join method https://doc.rust-lang.org/std/slice/trait.SliceConcatExt.html#tymethod.join, is it possible to use it?